src/switch_rtp.c

```diff
@@ -5528,25 +5530,38 @@ SWITCH_DECLARE(void) switch_rtp_break(switch_rtp_t *rtp_session)
 
 SWITCH_DECLARE(void) switch_rtp_kill_socket(switch_rtp_t *rtp_session)
 {
+	switch_socket_t *sock_input, *sock_output;
 	switch_assert(rtp_session != NULL);
 	switch_mutex_lock(rtp_session->flag_mutex);
 	if (rtp_session->flags[SWITCH_RTP_FLAG_IO]) {
 		rtp_session->flags[SWITCH_RTP_FLAG_IO] = 0;
 		if (rtp_session->sock_input) {
 			ping_socket(rtp_session);
-			switch_socket_shutdown(rtp_session->sock_input, SWITCH_SHUTDOWN_READWRITE);
+			sock_input = rtp_session->sock_input;
+			switch_socket_shutdown(sock_input, SWITCH_SHUTDOWN_READWRITE);
```

Comment:
There is a scenario where sock_output points to the same address as sock_input. Shutting it down here will make sock_output points to garbage memory. You may need to check sock_output before destroying sock_input